plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/joinDsl.kt

```diff
@@ -1,13 +1,29 @@
 package org.jetbrains.kotlinx.dataframe.plugin.impl.api
 
+import org.jetbrains.kotlinx.dataframe.api.ColumnMatch
+import org.jetbrains.kotlinx.dataframe.columns.ColumnResolutionContext
+import org.jetbrains.kotlinx.dataframe.columns.ColumnWithPath
 import org.jetbrains.kotlinx.dataframe.plugin.impl.AbstractInterpreter
 import org.jetbrains.kotlinx.dataframe.plugin.impl.Arguments
+import org.jetbrains.kotlinx.dataframe.plugin.impl.PluginDataFrameSchema
+import org.jetbrains.kotlinx.dataframe.plugin.impl.data.ColumnWithPathApproximation
 
-internal data class ColumnMatchApproximation(val left: ColumnsResolver, val right: ColumnsResolver)
+internal data class ColumnMatchApproximation(
```

Comment:
Could you give me a hint, what does Approximation mean. I see a lot in naming, but could not understand the idea

Reply:
In library we have ColumnMatch, and operations work with this class. We implement the same operation in compiler plugin, but we cannot cover 100% of usages, for example if some variable is used instead of constant. So it wouldn't be fair to just name it "ColumnMatch". Approximation indicates that it's "compile-time ColumnMatch". Something like this